tests/core/ast_transforms/test_basic_sanity.py

```diff
@@ -43,6 +56,46 @@ def test_child_transform_square_to_quadruple():
     assert final == 36
 
 
+def test_child_transform_square_to_zero():
+    original = parent(2)
+    assert original == 16
+
+    assert zero("ignored") == 0
+
+    result = parent(2, child=zero)
+    assert result == 0
+
+    final_two = parent(2)
+    assert final_two == 16
+
+    final = parent(3)
+    assert final == 36
+
+
+def test_problematic_child_transform_with_faked_child():
+    # function with a faked name should be respected as an override
+    # TODO this test captures current problematic behaviour which should be fixed
```

Comment:
I'm still not convinced by this - faking a function name is always going to mess up a jit compiler, so i don't see this as a valid use case. I think we should just remove this test.

Reply:
I think we should document the library's shortcomings. Tests are good way of capturing this - "you don't have to take our word for it, just take a look at the test". This also encourages contribution, to address "known issues".